src/redprl/signature_mini.sml

```diff
@@ -96,7 +96,8 @@ struct
           case psort of 
             RedPrlSortData.HYP tau =>
               let
-                val RedPrlParameterTerm.VAR v = ptm
+                val v = case ptm of RedPrlParameterTerm.VAR v => v
+                           | _ => raise Fail "Hypothesis renaming failed: not a variable"
```

Comment:
Let's keep this one, since this is a place where I could imagine a bug happening. Your error message looks good.